packages/react/avatar/src/Avatar.tsx

```diff
@@ -63,16 +63,16 @@ const AvatarImage = React.forwardRef((props, forwardedRef) => {
     onLoadingStatusChange: onLoadingStatusChangeProp = () => {},
     ...imageProps
   } = props;
-  const [, setImageLoadingStatus] = useAvatarContext(IMAGE_NAME);
+  const { onImageLoadingStatusChange } = useAvatarContext(IMAGE_NAME);
   const imageLoadingStatus = useImageLoadingStatus(src);
   const onLoadingStatusChange = useCallbackRef(onLoadingStatusChangeProp);
 
   useLayoutEffect(() => {
     if (imageLoadingStatus !== 'idle') {
       onLoadingStatusChange(imageLoadingStatus);
-      setImageLoadingStatus(imageLoadingStatus);
+      onImageLoadingStatusChange(imageLoadingStatus);
```

Comment:
I saw that rename in the other one too (was it `Accordion`? I think I've approved and merged that one).
Seeing it again now I'm not sure I'm fully onboard with it, why rename to and `onChange` callback, when really it's a state setter? I feel like it makes it less readable too for me, as it I can't see anymore that I'm doing `context.setSomething` which indicates I'm syncing something back up.

Reply:
It felt weird to me to add methods to a component (`AccordionProvider`) that don't follow our callback naming convention. I see context like private props here and we don't have props `setOpen` props. 

We name our callbacks in a way that communicates what event occurred, not what the method does with the value (implementation details). It doesn't feel necessary for a child component to know the implementation detail of the context, it just needs to communicate to context that it changed and it's up to the context method to decide what to do with it. It might set state today, it might set a ref tomorrow but the API shouldn't need to change if that happens.